### PR TITLE
Add keyboard shortcuts to sleep panel

### PR DESCRIPTION
**Pull request by gaul**
This allows the up and down arrows to select a sleep duration and
enter to confirm it.  It also works around a focus regression in
Firefox Nightly 93 and matches the Multi-Account Containers behavior.
References #488.  References #32.

### CODE REVIEW

File: src/popup/snooze.html [hunk 1]
```diff
@@ -6,7 +6,21 @@
...
+      <div class="panel-wrapper">
```
Comment (bwinton) [inline src/popup/snooze.html:9]: I wonder why it was all on one line previously… Perhaps something about extra whitespace messing with the styling? Could you add a comment with screenshots from before and after this patch is applied?

File: src/popup/snooze-content.js [hunk 2]
```diff
@@ -126,13 +126,44 @@ const resizeHandler = () => {
...
+  switch (e.keyCode) {
+  case 40: {
+    const nextElement = selectables[index + 1];
+    if (nextElement) {
+      nextElement.focus();
+    }
+    break;
+  }
+  case 38: {
+    const previousElement = selectables[index - 1];
```
Comment (bwinton) [inline src/popup/snooze-content.js:143]: Perhaps, if index is already negative, we should move to the bottom element, and have them wrap? What do you think?

**Review by bwinton — CHANGES_REQUESTED** [review]
Seems mostly good, but there are a couple of things I'd like to get fixed before I merge it…

**Comment by bwinton** [comment]
I also just noticed that the up/down arrows will continue to do stuff when the calendar widget is open, which they shouldn't. Maybe we should do the processing in `lib/components/ManagePanel.js` instead, to make sure it's scoped correctly…

**Comment by bwinton** [comment]
(Also, this kinda fixes #32, which is nice! 😃)

File: src/popup/snooze-content.js [hunk 3]
```diff
@@ -126,13 +126,44 @@ const resizeHandler = () => {
   });
 };
 
+const keyboardHandler = (e) => {
+  const selectables = [...document.querySelectorAll('.option')];
+  const element = document.activeElement;
+  const index = selectables.indexOf(element) || 0;
+
+  switch (e.keyCode) {
+  case 40: {
```
[the remark this answers is not in the document]
Comment (gaul) [reply src/popup/snooze-content.js:135]: Done.

File: src/popup/snooze-content.js [hunk 4]
```diff
@@ -126,13 +126,44 @@ const resizeHandler = () => {
...
+    break;
+  }
+  case 38: {
+    const previousElement = selectables[index - 1];
+    if (previousElement) {
+      previousElement.focus();
+    }
+    break;
+  }
+  case 39:
```
[the remark this answers is not in the document]
Comment (gaul) [reply src/popup/snooze-content.js:149]: Done.

File: src/popup/snooze-content.js (hunk 2 above)
Comment (gaul) [reply src/popup/snooze-content.js:143]: Done.

File: src/popup/snooze.html (hunk 1 above)
Comment (gaul) [reply src/popup/snooze.html:9]: Removed unneeded change.

**Comment by gaul** [comment]
> I also just noticed that the up/down arrows will continue to do stuff when the calendar widget is open, which they shouldn't. Maybe we should do the processing in `lib/components/ManagePanel.js` instead, to make sure it's scoped correctly…

Could you give me some more guidance on this?  I don't understand where I would add the keyboard handler.

**Comment by bwinton** [comment]
> > I also just noticed that the up/down arrows will continue to do stuff when the calendar widget is open, which they shouldn't. Maybe we should do the processing in `lib/components/ManagePanel.js` instead, to make sure it's scoped correctly…
> 
> Could you give me some more guidance on this? I don't understand where I would add the keyboard handler.

Yeah, so on [this line](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L70), there's an `onKeyPress` handler, which forwards all the keys to [this function](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L182-L188). So maybe we can add an `onKeyPress` to the div on [this line](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L45), and handle the keys there? (We'd need to allow them to pass through [this function](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L178-L180), but that shouldn't be so bad…

**Comment by gaul** [comment]
> Yeah, so on [this line](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L70), there's an `onKeyPress` handler, which forwards all the keys to [this function](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L182-L188). So maybe we can add an `onKeyPress` to the div on [this line](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L45), and handle the keys there? (We'd need to allow them to pass through [this function](https://github.com/bwinton/SnoozeTabs/blob/main/src/lib/components/ManagePanel.js#L178-L180), but that shouldn't be so bad…

Sorry, I spent a few hours on this and probably won't figure it out myself.  I suspect you mean `MainPanel.js` though?

**Review by bwinton — APPROVED** [review]
Well, let's not let the perfect be the enemy of the good.
👍🏼 from me, and we can move the code around later. 🙂

**Pull request merged**